priam/src/main/java/com/netflix/priam/backupv2/BackupTTLTask.java

```diff
@@ -260,18 +261,18 @@ public static TaskTimer getTimer(
     private class MetaFileWalker extends MetaFileReader {
         @Override
         public void process(ColumnFamilyResult columnfamilyResult) {
-            columnfamilyResult
-                    .getSstables()
-                    .forEach(
-                            ssTableResult ->
-                                    ssTableResult
-                                            .getSstableComponents()
-                                            .forEach(
-                                                    fileUploadResult ->
-                                                            filesInMeta.put(
-                                                                    fileUploadResult
-                                                                            .getBackupPath(),
-                                                                    null)));
+            for (ColumnFamilyResult.SSTableResult sstable : columnfamilyResult.getSstables()) {
+                for (FileUploadResult component : sstable.getSstableComponents()) {
+                    filesInMeta.put(removeCompressionPart(component.getBackupPath()), null);
+                }
+            }
+        }
+    }
+
+    private String removeCompressionPart(String backupPath) {
```

Comment:
this can have some easy unit tests, would be helpful documentation a bit too on some expected paths look like